### PR TITLE
funcParser_tool: v1.0

### PR DESCRIPTION
**Pull request by stefanodellosa**
funcParser tool is a PoC developed to support the FFI activities.
It parses source files and extracts relevant information, such as
function definitions, function calls, function pointers, etc.
All information are then dumped into a SQLite db for further
processing.

Signed-off-by: Stefano Dell'Osa <stefano.dellosa@intel.com>

### CODE REVIEW

File: funcParser_tool/LICENSE [hunk 1]
```diff
@@ -0,0 +1,72 @@
+GNU GENERAL PUBLIC LICENSE
+Version 2, June 1991
+
+Copyright (C) 1989, 1991 Free Software Foundation, Inc.
```
Comment (reiterative) [inline funcParser_tool/LICENSE]: This should be the copyright date and name of author for *this* software, as in the source code files.

Comment (stefanodellosa) [reply funcParser_tool/LICENSE]: Good catch! Updated.

Comment (bulwahn) [reply funcParser_tool/LICENSE]: Sorry, that was wrong. This is the copyright statement for the copyright holder of the GNU GENERAL PUBLIC LICENSE text. I doubt that FSF has transfer the copyright of this license text to Intel.

**Comment by bulwahn** [comment]
Please make the license information follow the https://reuse.software/ guidelines.
Other than that, this looks good. I am just wondering if you are really interested in implementing a proper static analysis tool by just using a few regular expressions in python to create a C language parser, or if this is just intended as an explorative tool to be eventually replaced by one tool among all the pre-existing tools with the same functionality.

File: funcParser_tool/LICENSE (hunk 1 above)
Comment (stefanodellosa) [reply funcParser_tool/LICENSE]: Restored to original statement.

**Comment by stefanodellosa** [comment]
> Please make the license information follow the https://reuse.software/ guidelines.
> Other than that, this looks good. I am just wondering if you are really interested in implementing a proper static analysis tool by just using a few regular expressions in python to create a C language parser, or if this is just intended as an explorative tool to be eventually replaced by one tool among all the pre-existing tools with the same functionality.

Thanks for your comment! I've updated the licensing information according to REUSE guidelines. Let me know if you still find gaps.

With respect to the parser, as I said in the last WG, funcParser tool is developed as PoC in order to show how to extract some FFI relevant information from source code to the WG audience. It has been then released to the community to make it publicly available.

The goal is definitely not to re-invent the wheel but on the other side there might be aspects of FFI that are not captured by a static analysis tool (perhaps not in the form we need for FFI scope), and static analysis aspects that funcParser does not covers since not in scope for FFI. 

Said that, we can definitely discuss about alternatives static analysis tools that can be used instead, or whether it is useful to have a custom tool that implements specifically designed features to support FFI. I am open to that.

**Review by paolonig — APPROVED** [review]
looks good to me

**Pull request merged**